lib/ddtrace/metrics.rb

```diff
@@ -244,19 +248,32 @@ def gauge(stat, value, options = nil)
 
     private
 
+    def dogstatsd_version
+      return @dogstatsd_version if instance_variable_defined?(:@dogstatsd_version)
+
+      @dogstatsd_version = (
+        defined?(Datadog::Statsd::VERSION) &&
+          Datadog::Statsd::VERSION &&
+          Gem::Version.new(Datadog::Statsd::VERSION)
+      ) || (
+        Gem.loaded_specs['dogstatsd-ruby'] &&
+          Gem.loaded_specs['dogstatsd-ruby'].version
+      )
+    end
+
     INCOMPATIBLE_STATSD_ONLY_ONCE = Datadog::Utils::OnlyOnce.new
     private_constant :INCOMPATIBLE_STATSD_ONLY_ONCE
 
     # IMPORTANT: Once you delete this, you probably want to update the `#ignored_statsd_warning` below where we
     # recommend that customers add `gem 'dogstatsd-ruby', '~> 4.0'` to their Gemfile to perhaps state something
     # different.
     def incompatible_statsd_warning
-      return if Gem.loaded_specs['dogstatsd-ruby'].version < Gem::Version.new('5.0')
+      return if dogstatsd_version < Gem::Version.new('5.0') || dogstatsd_version >= Gem::Version.new('5.2')
 
       INCOMPATIBLE_STATSD_ONLY_ONCE.run do
         Datadog.logger.warn(
-          'This version of `ddtrace` is incompatible with `dogstastd-ruby` version >= 5.0 and can ' \
-          'cause unbounded memory usage. Please use `dogstastd-ruby` version < 5.0 instead.'
+          '`ddtrace` is incompatible with `dogstastd-ruby` versions 5.0.0, 5.0.1, and 5.2.0 and can ' \
+          'cause unbounded memory usage. Use `dogstastd-ruby` version >= 5.2 instead.'
         )
       end
     end
```

Comment:
Note: In #1575 we had discussed and decided to remove this, and instead just make ddtrace treat 5.0.0, 5.0.1 and 5.1.0 as "incompatible" versions and outright not use them. See the discussion in the PR for details.

Reply:
I thought it wasn't a good idea to make it incompatible, but I now agree with metrics being missing for safety. Changes applied.